src/Gaffer/FileSystemPath.cpp

```diff
@@ -542,3 +551,119 @@ PathFilterPtr FileSystemPath::createStandardFilter( const std::vector<std::strin
 
 	return result;
 }
+
+// Create a path that is aware of differences in OS naming schemes.
+// Linux and MacOS can have a relative path with no root, or a path rooted at "/" :
+// Windows can have a relative path with no root, a path rooted at a drive letter followed by a colon
+// or a path rooted at a server name and share name (UNC paths).
+
+void FileSystemPath::setFromString(const std::string &string)
+{
+	Names newNames;
+	std::string sanitizedString = string;
+
+	boost::replace_all( sanitizedString, g_windowsSeparator.c_str(), g_genericSeparator.c_str() );
+
+#ifdef _MSC_VER
+	// If `string` is coming from a PathMatcher, it will always have a single leading slash.
+	// On Windows, interpret that to be the start of a UNC path by adding an addition leading slash
+	// If it is in fact a drive letter path, the extra slash won't cause problems in
+	// the conditionals below with `newNames`.
+	if(
+		sanitizedString.size() &&
+		sanitizedString[0] == '/' &&
+		!boost::starts_with( sanitizedString, g_uncPrefix.string() )
+	)
+	{
+		sanitizedString = g_genericSeparator.string() + sanitizedString;
+	}
+#endif
+
+	StringAlgo::tokenize<InternedString>(sanitizedString, '/', back_inserter(newNames));
+
+	InternedString newRoot;
+	if ( newNames.size() && boost::regex_match( newNames[0].string(), g_driveLetterPattern ) )
+	{
+		newRoot = newNames[0];
+		newNames.erase( newNames.begin() );
+	}
+	else if ( boost::starts_with( sanitizedString, g_uncGenericPrefix.string() ) )
+	{
+		// The server name and share name are both required before the path is valid.
+		if( newNames.size() > 1 )
```

Comment:
I went with `boost::filesystem` because `std::filesystem` isn't 100% drop-in compatible with `boost::filesystem` and it seems cleaner to stick with one scheme throughout the file.

Reply:
Do you see any differences between `boost` and `std` that would prevent us moving to `std` easily in future?

Reply:
Nothing major I don't think - `status_error` and `file_not_found` don't seem to have a direct equivalent in `std::filesystem`. I haven't dug around too much to see what a replacement would be. They are used at
https://github.com/GafferHQ/gaffer/blob/4fa64dff9e097ba3a2979b657ea12baa0c5a0ba9/src/Gaffer/FileSystemPath.cpp#L95-L109